publisher/elasticsearch.go

```diff
@@ -0,0 +1,221 @@
+package publisher
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	retryhttp "github.com/hashicorp/go-retryablehttp"
+)
+
+// ElasticSearchClient holds config for Kafkarest target
+type ElasticSearchClient struct {
+	// Host for elasticsearch
+	Host string `yaml:"host" json:"host"`
+
+	// Port for elasticsearch
+	Port string `yaml:"port" json:"port"`
+
+	// Protocol to connect to elasticsearch
+	Protocol string `yaml:"protocol" json:"protocol"`
+
+	// Index to send data to
+	Index string `yaml:"index" json:"index"`
+
+	// Username if basic auth is used
+	Username string `yaml:"user,omitempty" json:"user,omitempty"`
+
+	// Password if basic auth is used
+	Password string `yaml:"password,omitempty" json:"password,omitempty"`
+
+	// ES_7x to know if target is elastic version 7.x
+	ES_7x bool `yaml:"es_7x" json:"es_7x"`
```

Comment:
Do you need this as input? check if you can query to ES to get version and set this flag. sf-trace-server is likely getting the version on the fly.

Reply:
will add logic to get version in code itself